examples/call-evm-counter/src/contract.rs

```diff
@@ -38,18 +49,23 @@ impl Contract for CallCounterContract {
     }
 
     async fn execute_operation(&mut self, operation: CallCounterOperation) -> u64 {
-        let CallCounterOperation::Increment(increment) = operation;
         sol! {
             function increment(uint64 input);
+            function call_from_wasm(address remote_address);
+        }
+        match operation {
+            CallCounterOperation::Increment(increment) => {
+                let operation = incrementCall { input: increment };
+                let operation = operation.abi_encode();
+                self.process_operation(operation)
+            }
+            CallCounterOperation::TestCallAddress => {
+                let remote_address = address!("0000000000000000000000000000000000000000");
```

Comment:
I think it's quite non-intuitive that `msg.sender` can be so many different values:
1. if EVM contract/account makes a contract call - we set it to the correct address
2. if we make a service call then it's `0x000...20`
3. if Linera contract/account makes the call - we set it to `0x000..00`.